src/__tests__/parser.test.ts

```diff
@@ -0,0 +1,457 @@
+      const file = stripIndent`
+      # Test file
+      
+      # description: your Twilio phone number
+      # format: phone_number
+      # required: true
+      TWILIO_PHONE_NUMBER=+12223334444
+
+      # description: Message to play
+      # required: false
+      MESSAGE=
+    `;
+      const result = parse(file);
+      expect(result.variables).toEqual([
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_PHONE_NUMBER',
+          format: 'phone_number',
+          required: true,
+          description: 'your Twilio phone number',
+          default: '+12223334444',
+        },
+        {
+          ...DEFAULT_ENTRY,
+          key: 'MESSAGE',
+          required: false,
+          description: 'Message to play',
+          default: null,
+        },
+      ]);
+    });
+
+    test('recognizes link comment', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: sid
+      # link: https://www.twilio.com/console
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      # link: https://www.twilio.com/console
+      TWILIO_AUTH_TOKEN=
+    `;
+      const result = parse(file);
+      expect(result.variables).toEqual([
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_ACCOUNT_SID',
+          description: 'Your Twilio Account SID',
+          format: 'sid',
+          link: 'https://www.twilio.com/console',
+          default: 'ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+        },
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_AUTH_TOKEN',
+          description: 'Your Twilio Auth Token',
+          format: 'secret',
+          link: 'https://www.twilio.com/console',
+          default: null,
+        },
+      ]);
+    });
+
+    test('recognizes default comment', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: sid
+      # link: https://www.twilio.com/console
+      # default: ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      TWILIO_ACCOUNT_SID=
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      # link: https://www.twilio.com/console
+      TWILIO_AUTH_TOKEN=
+    `;
+      const result = parse(file);
+      expect(result.variables).toEqual([
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_ACCOUNT_SID',
+          description: 'Your Twilio Account SID',
+          format: 'sid',
+          link: 'https://www.twilio.com/console',
+          default: 'ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+        },
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_AUTH_TOKEN',
+          description: 'Your Twilio Auth Token',
+          format: 'secret',
+          link: 'https://www.twilio.com/console',
+          default: null,
+        },
+      ]);
+    });
+
+    test('recognizes configurable comment', () => {
```

Comment:
and an invalid case here too

Reply:
think this could still use a quick invalid case to ensure it throws appropriately